src/app/core/services/ess-upload-file.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+import { AppConfigService } from './app-config.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EssUploadFileService {
+  private validEncs: string[];
+  private maxEncLimit: number;
+  constructor() {
+    this.maxEncLimit = AppConfigService.settings['essConfig'].MaxEncLimit;
+  }
+
+  validateCSVFile() {}
+
+  isValidEncFile(encFileTYpe: string, encList: string[]): boolean {
+    if (
+      encFileTYpe === 'text/plain' &&
+      encList[2] === ':ENC' &&
+      encList[encList.length - 1] === ':ECS'
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  validateENCFormat(encName: string) {
+    const pattern = /[A-Z]{2}[1-68][A-Z0-9]{5}$/;
+    return encName.match(pattern);
+  }
+
+  extractEncsFromFile(fileType: string, processedData: string[]) {
```

Comment:
Please remove the duplicated file validation code from `extractEncsFromFile`. The file structure has already been validated by `isValidEncFile` so doesn't need to be repeated again